src/bin/languageserver/mod.rs

```diff
@@ -1282,6 +1726,77 @@ fn loc_to_range(loc: &pt::Loc, file: &ast::File) -> Range {
     Range::new(start, end)
 }
 
+fn get_file_no_from_loc(loc: &pt::Loc) -> Option<usize> {
+    if let pt::Loc::File(fno, _, _) = loc {
+        Some(*fno)
+    } else {
+        None
+    }
+}
+
+fn get_type_definition(ty: &Type) -> Option<DefinitionIndex> {
+    match ty {
+        Type::Enum(id) => Some(DefinitionIndex::Enum(*id)),
+        Type::Struct(StructType::UserDefined(id)) => Some(DefinitionIndex::Struct(*id)),
+        Type::Array(ty, _) => get_type_definition(ty),
+        Type::Ref(ty) => get_type_definition(ty),
+        Type::StorageRef(_, ty) => get_type_definition(ty),
+        Type::Contract(id) => Some(DefinitionIndex::Contract(*id)),
+        Type::UserType(id) => Some(DefinitionIndex::UserType(*id)),
+        _ => None,
+    }
+}
+
+fn remove_reference(ty: Type) -> Type {
+    match ty {
+        Type::Ref(ty) => *ty,
+        Type::StorageRef(_, ty) => *ty,
+        ty => ty,
+    }
+}
```

Comment:
`ty.deref_any()`

Reply:
Thanks! Replaced it with `deref_any` function